infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/LogReplicationFSM.java

```diff
@@ -223,10 +226,82 @@ public class LogReplicationFSM {
     public LogReplicationFSM(CorfuRuntime runtime, LogReplicationConfigManager configManager, DataSender dataSender,
                              ReadProcessor readProcessor, ExecutorService workers, LogReplicationAckReader ackReader,
                              LogReplicationConfigManager tableManagerPlugin, ReplicationSession replicationSession) {
-        // Use stream-based readers for snapshot and log entry sync reads
-        this(runtime, new StreamsSnapshotReader(runtime, configManager, replicationSession), dataSender,
-            new StreamsLogEntryReader(runtime, configManager, replicationSession), readProcessor, configManager,
-            workers, ackReader, tableManagerPlugin, replicationSession);
+
+        this.snapshotReader = createSnapshotReader(runtime, configManager, replicationSession);
+        this.logEntryReader = createLogEntryReader(runtime, configManager, replicationSession);
+        this.ackReader = ackReader;
+        this.tableManagerPlugin = tableManagerPlugin;
```

Comment:
I also think the 2 constructors in this class could be refactored/cleanup a little bit